actix-http/src/h1/service.rs

```diff
@@ -417,47 +372,27 @@ where
     type Future = Dispatcher<T, S, B, X, U>;
 
     fn poll_ready(&self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        let ready = self
-            .flow
-            .expect
-            .poll_ready(cx)
-            .map_err(|e| {
+        ready!(self.flow.expect.poll_ready(cx)).map_err(|e| {
+            let e = e.into();
+            log::error!("Http expect service readiness error: {:?}", e);
+            DispatchError::Service(e)
+        })?;
+
+        if let Some(ref upg) = self.flow.upgrade {
+            ready!(upg.poll_ready(cx)).map_err(|e| {
                 let e = e.into();
-                log::error!("Http service readiness error: {:?}", e);
+                log::error!("Http upgrade service readiness error: {:?}", e);
                 DispatchError::Service(e)
-            })?
-            .is_ready();
-
-        let ready = self
-            .flow
-            .service
-            .poll_ready(cx)
-            .map_err(|e| {
-                let e = e.into();
-                log::error!("Http service readiness error: {:?}", e);
-                DispatchError::Service(e)
-            })?
-            .is_ready()
-            && ready;
-
-        let ready = if let Some(ref upg) = self.flow.upgrade {
-            upg.poll_ready(cx)
-                .map_err(|e| {
-                    let e = e.into();
-                    log::error!("Http service readiness error: {:?}", e);
-                    DispatchError::Service(e)
-                })?
-                .is_ready()
-                && ready
-        } else {
-            ready
+            })?;
         };
 
-        if ready {
-            Poll::Ready(Ok(()))
-        } else {
-            Poll::Pending
-        }
+        ready!(self.flow.service.poll_ready(cx)).map_err(|e| {
+            let e = e.into();
+            log::error!("Http service readiness error: {:?}", e);
+            DispatchError::Service(e)
+        })?;
```

Comment:
old order: expect => service => upgrade  
new order: expect => upgrade => service

do you expect that to have any impact?

Reply:
I don't think it make any difference. All service type in HttpFlow must be ready(If upgrade service is Some) before accepting a request.